File: commands.js
var riskofrain2 = require('./riskofrain2.js');
var mcRCON = require('./mcRCON.js');
var https = require('https');
var fileManager = require('./fileManager.js');
var randomMod = require('./randomMod.js');
var members = require('./members.js');
const masterColor = 7871916

const command = function(client, message, channel, lastNum) 
{
  var args = message.content.substring(1).split(' ');
  var cmd = args[0];
  console.log("Command issued " + message.content) 

  args = args.splice(1);

  if (message.content.toLowerCase().includes("anime".toLowerCase()) == true) {
    var path = "./images/anime/"
    if ((message.content.substring(0, 7) == "!anime_".toLowerCase()) && (parseInt(message.content.substring(6))) != NaN) {
        console.log("Trying to find anime_" + parseInt(message.content.substring(7)))
        var num = parseInt(message.content.substring(7))
        var fileName = "anime_"
        channel.send(fileManager.sendImage(num, path, fileName, message, masterColor))
    } else {
      var num = Math.floor(Math.random() * fileManager.getAllDirFiles("./images/anime").length)
      console.log(fileManager.getAllDirFiles("./images/anime").length)
      var fileName = "anime_"
      if (num == 5) {
        console.log("Rerolling...")
        num = Math.floor(Math.random() * fileManager.getAllDirFiles("./images/anime").length)
      }
      channel.send(fileManager.sendImage(num, path, fileName, message, masterColor))
    }  
  }

  if (message.content.toLowerCase().includes("selfie".toLowerCase()) == true) {
    var path = "./images/selfies/"
    if ((message.content.substring(0, 8) == "!selfie_".toLowerCase()) && (parseInt(message.content.substring(8))) != NaN) {
        console.log("Trying to find selfie_" + parseInt(message.content.substring(8)))
        var num = parseInt(message.content.substring(8))
        var fileName = "selfie_"
        channel.send(fileManager.sendImage(num, path, fileName, message, masterColor))
    } else {
      var num = Math.floor(Math.random() * fileManager.getAllDirFiles("./images/selfies").length)
      console.log(fileManager.getAllDirFiles("./images/selfies").length)
      while (num == lastNum) {
        num = Math.floor(Math.random() * fileManager.getAllDirFiles("./images/selfies").length)
      }
      lastNum = num
      var fileName = "selfie_"
      channel.send(fileManager.sendImage(num, path, fileName, message, masterColor))
    }  
  }

  if (message.content.toLowerCase().includes("!broadcast".toLowerCase()) == true) {
    message.content = message.content.substring(10)  

    client.channels.cache.get('254779349352448001').send(message.content);
  }
 
  if (message.content.toLowerCase().includes('who is'.toLowerCase()) == true){
      if (message.content.toLowerCase().includes('your mother'.toLowerCase()) == true) {
          message.channel.send('<@!256343280285908992> is my mother');
      } else if (message.content.toLowerCase().includes('your father'.toLowerCase()) == true) {
        message.channel.send('<@!253710834553847808> is my father');
      } else if (message.content.toLowerCase().includes('step mother'.toLowerCase()) == true) {
          message.channel.send('<@!280117740222676992> is my step mother');
      } else if (message.content.toLowerCase().includes('your brother'.toLowerCase()) == true) {
          message.channel.send('<@!251488731750465536> is my brother');
      } else {
          message.channel.send('Who?');
      }
  } else if (message.content.toLowerCase().includes('roll_d'.toLowerCase()) == true) { 
    var dice = 1
    dice = parseInt(cmd.substring(6))
    var num = Math.floor(Math.random() * dice) + 1;
    var embed = {
      "embed": {
        "description": `You rolled ${num} on your d${dice}. Good job! At the very least you get an A+ for effort so isn't that nice.`,
        "color": 2464068,
        "thumbnail": {
          "url": "https://gilkalai.files.wordpress.com/2017/09/dice.png?w=640"
        },
        "author": {
          "name": "Steve from Accounting",
          "url": "https://www.google.com/error",
          "icon_url": "https://www.topaccountingdegrees.org/wp-content/uploads/2015/08/Accounting-7.jpg"
        }
      }
    };
    channel.send(embed);
  } else if (message.content.toLowerCase().includes('sins'.toLowerCase()) == true) { 
    if (!message.mentions.users.size) {
      message.channel.send('You need to tag someone to get their sins...');
    } else {
      var member;
      const taggedUser = message.mentions.users.first();
      console.log(`Sins of: ${taggedUser.username}`)
      member = members.checkMember(taggedUser.username, taggedUser.id)
      console.log(`Sins: ${member}`)
      var data = {
        "to": channel,
        "embed": {
          "title": `Sins of ${taggedUser.username}`,
          "description": member.sins,
          "color": 7871916,
          "thumbnail": {
            "url": taggedUser.displayAvatarURL({ format: "png", dynamic: true }),
          }
        }
      };
      channel.send(data);
    }
    
  } else {
    var url = "";
      switch(cmd) {
          case 'ping':
            message.channel.send('What are you expecting? Me to say pong back?');
          break;
          case 'wtf':
            message.channel.send('Rude!');
          break;
          case 'nani':
            message.channel.send('何');
          break;
          case 'lastNum':
            message.channel.send(("The last gif number was: " + lastNum));
          break;
          case 'random':
            var text = randomMod.randomCommands()
            message.channel.send((text));
          break;
          case 'birthdayReset':
              birthdayLastDate = new Date(2019,3,31)
              message.channel.send("Cooldown reset");
          break;
          case 'thispersondoesnotexist':
            var plScraper = require('./pl-scraper')
            url = "https://thispersondoesnotexist.com/";
            var filePath = "./images/thispersondoesnotexist/"
            var name = "tpdne.jpg"
            console.log(filePath + name)
            plScraper.makeScrape(url, filePath, name);
            channel.send({
              files: [(filePath + name)]
            });
          break;
            case 'waifu':
            var num = Math.floor(Math.random() * 100000)
            url = "https://www.thiswaifudoesnotexist.net/example-" + num + ".jpg";
            var filePath = "./images/waifu/"
            var name = "waifu"
            console.log(url)

            var embed = {
              "image": {
                  "url": url
                }
              };
          case 'waifu':
            var num = Math.floor(Math.random() * 100000)
            url = "https://www.thiswaifudoesnotexist.net/example-" + num + ".jpg";
            var filePath = "./images/waifu/"
            var name = "waifu"
            console.log(url)

            var embed = {
              "image": {
                  "url": url
                }
              };
          message.channel.send({ embed });
          break;
          case 'hot':
            console.log("HOT")
            channel.send({
              files: ['./video/Hot.mp4']
            });
          break;
          case 'joke':
            //Info at: https://sv443.net/jokeapi/v2/
            url = "https://sv443.net/jokeapi/v2/joke/Any?blacklistFlags=racist&type=single";

            https.get(url, res => {
              res.setEncoding("utf8");
              let body = "";
              res.on("data", data => {
                body += data;
              });
              res.on("end", () => {
                body = JSON.parse(body);
                console.log(body);
                console.log(body.joke)
                var text = body.joke
                channel.send(body.joke);
              });
            });
          break;
          case 'insultme':
            //Info at: https://evilinsult.com/api/
            url = "https://evilinsult.com/generate_insult.php?lang=en&type=json";

            https.get(url, res => {
              res.setEncoding("utf8");
              let body = "";
              res.on("data", data => {
                body += data;
              });
              res.on("end", () => {
                body = JSON.parse(body);
                console.log(body);
                console.log(body.insult)
                var text = body.insult
                channel.send(body.insult);
              });
            });
            break;
          case 'yesorno':
            //Info at: https://yesno.wtf/api
            url = "https://yesno.wtf/api";

            https.get(url, res => {
              res.setEncoding("utf8");
              let body = "";
              res.on("data", data => {
                body += data;
              });
              res.on("end", () => {
                body = JSON.parse(body);
                console.log(body);
                console.log(body.answer)
                var text = body.answer
                channel.send(body.answer);
              });
            });
          break;
          case 'mcday':
            console.log("Gettting Server Stat")
            mcRCON.sendCommand('time set day', channel)
          break;
          case 'mcnight':
            console.log("Gettting Server Stat")
            mcRCON.sendCommand('time set night', channel)
          break;
          case 'mctime':
            console.log("Gettting Server Stat")
            mcRCON.sendCommand('time world', channel)
          break;
          case 'mcseed':
            console.log("Gettting Server Stat")
            mcRCON.sendCommand('seed', channel)
          break;
          case 'mcinfo':
            console.log("Gettting Server Stat")
            mcRCON.LookServerFL(channel)
          break;
          case 'fact':
            //Info at: https://uselessfacts.jsph.pl/
            url = "https://uselessfacts.jsph.pl/random.json?language=en";

            https.get(url, res => {
              res.setEncoding("utf8");
              let body = "";
              res.on("data", data => {
                body += data;
              });
              res.on("end", () => {
                body = JSON.parse(body);
                console.log(body);
                console.log(body.text)
                var text = body.text
                var embed = {
                    "embed": {
                      "description": `Fact: ${text}\nSource: [${body.source}](${body.source_url})`,
                      "url": "https://discordapp.com",
                      "color": 4550821,
                      "author": {
                        "name": "Dr. Fact",
                        "url": "https://uselessfacts.jsph.pl/",
                        "icon_url": "https://img.icons8.com/bubbles/2x/light-on.png"
                      }
                    }
                  };
                  channel.send(embed);
                });
              });
          break;
          case 'advise':
            //Info at: https://api.adviceslip.com/
            url = "https://api.adviceslip.com/advice";

            https.get(url, res => {
              res.setEncoding("utf8");
              let body = "";
              res.on("data", data => {
                body += data;
              });
              res.on("end", () => {
                body = JSON.parse(body);
                console.log(body);
                console.log(body.slip.advice)
                var text = body.slip.advice
                var embed = {
                    "embed": {
                      "description": `Advice: ${text}`,
                      "url": "https://discordapp.com",
                      "color": 4550821,
                      "author": {
                        "name": "Advice Co.",
                        "url": "https://api.adviceslip.com/",
                        "icon_url": "https://www.partnersinprojectgreen.com/wp-content/uploads/2017/01/PPG_EV_ICON_HELP.png"
                      }
                    }
                  };
                  channel.send(embed);
                });
              });
          break;
          case 'cat':
            url = "https://api.thecatapi.com/v1/images/search?api_key=b9a826e6-fac5-43e7-8af1-aa47523e1bbd";

            https.get(url, res => {
              res.setEncoding("utf8");
              let body = "";
              res.on("data", data => {
                body += data;
              });
              res.on("end", () => {
                body = JSON.parse(body);
                console.log(body);
                var data = body[0]
                console.log(data)
                url = data.url
                var embed = {
                  "image": {
                    "url": url
                  }
                };
                message.channel.send({ embed });
              });
            });
          break;
          case 'zerotier':
            embed = {
              "embed": {
                "title": "ZeroTier Info",
                "url": "https://www.zerotier.com/",
                "color": 16763904,
                "thumbnail": {
                  "url": "https://my.zerotier.com/img/zerotier-logo-white.png"
                },
                "author": {
                  "name": "ZeroTier",
                  "url": "https://discordapp.com",
                  "icon_url": "https://my.zerotier.com/img/zerotier-logo-white.png"
                },
                "fields": [
                  {
                    "name": "Step 1",
                    "value": "[Download ZeroTier](https://www.zerotier.com/download/) and install it."
                  },
                  {
                    "name": "Step 2",
                    "value": "Join: 8850338390b695d0 by right clicking the ZeroTier icon in the task bar."
                  },
                  {
                    "name": "Step 3",
                    "value": "DM Geoffery10 and let him know to let you into the server."
                  },
                  {
                    "name": "Step 4",
                    "value": "Wait patiently..."
                  }
                ]
              }
            }
            channel.send(embed);
          break;
          case 'r2loadout':
            // Start selection... Picking survivor
            var survivor = riskofrain2.randomLoadout('survivor')
            var white = riskofrain2.randomLoadout('white')
            var green = riskofrain2.randomLoadout('green')
            var red = riskofrain2.randomLoadout('red')
            var equipment = riskofrain2.randomLoadout('equipment')
            var yellow = riskofrain2.randomLoadout('yellow')
            var lunar = riskofrain2.randomLoadout('lunar')
            var lunarEquipment = riskofrain2.randomLoadout('lunarEquipment')
            
            var data = {
              "to": channel,
              "message": "Good luck survivor...",
              "embed": {
                "title": "Risk of Rain 2 - Random Loadout",
                "description": "Item info can be found at the [Risk of Rain 2 Wiki](https://riskofrain2.gamepedia.com/Items).",
                "color": 4726857,
                "footer": {
                  "icon_url": "https://static.wikia.nocookie.net/be327120-a2dd-4467-bf86-8c2212037668",
                  "text": "Good luck survivor..."
                },
                "thumbnail": {
                  "url": survivor
                },
                "fields": [
                  {
                    "name": "White",
                    "value": white
                  },
                  {
                    "name": "Green",
                    "value": green
                  },
                  {
                    "name": "Red",
                    "value": red
                  },
                  {
                    "name": "Equipment",
                    "value": equipment
                  },
                  {
                    "name": "Yellow",
                    "value": yellow
                  },
                  {
                    "name": "Lunar",
                    "value": lunar
                  },
                  {
                    "name": "Lunar Equipment",
                    "value": lunarEquipment
                  }
                ]
              }
            };
            channel.send(data);
          break;
          case 'help':
            console.log("Sending help!")
            var data = {
                    to: channel,
                    "embed": {
                      "title": "Commands",
                      "description": "This is a list of all the commands I can do:",
                      "fields": [
                        {
                          "name": "hentai",
                          "value": "That's illegal!"
                        },
                        {
                          "name": "owo or uwu",
                          "value": "owo"
                        },
                        {
                          "name": "sauce",
                          "value": "DANGER NSFW"
                        },
                        {
                          "name": "heresy",
                          "value": "Yikes that bad?"
                        },
                        {
                          "name": "ravioli ravioli",
                          "value": "DON'T LEWD!"
                        },
                        {
                          "name": "hello there",
                          "value": "It's a star wars reference"
                        },
                        {
                          "name": "trap",
                          "value": "Danger!"
                        },
                        {
                          "name": "!ping",
                          "value": "Pong probably"
                        },
                        {
                          "name": "!anime",
                          "value": "Anime gif\nIf send \"!anime_<number>\" then you can call a specific gif."
                        },
                        {
                          "name": "!r2loadout",
                          "value": "Gives you a complete random Risk of Rain 2 loadout to use in your next Artifact of Command run."
                        },
                        {
                          "name": "!mctime",
                          "value": "The time in the Minecraft Server (if running)."
                        },
                        {
                          "name": "!mcseed",
                          "value": "The seed of the Minecraft Server (if running)."
                        },
                        {
                          "name": "!wtf",
                          "value": "That's mean!"
                        },
                        {
                          "name": "!hot",
                          "value": "HOT!"
                        },
                        {
                          "name": "!roll_d<Your Number Here>",
                          "value": "Roll any dice. Example: !roll_d20"
                        },
                        {
                          "name": "!nani",
                          "value": "What?"
                        },
                        {
                          "name": "!fact",
                          "value": "I'll tell you a random fact."
                        },
                        {
                          "name": "!advise",
                          "value": "I'll tell you a some advice if you are that desperate..."
                        },
                        {
                          "name": "!joke",
                          "value": "I'll tell you a joke."
                        },
                        {
                          "name": "!insultme",
                          "value": "I'll tell you a different kind of joke..."
                        },
                        {
                          "name": "!yesorno",
                          "value": "I'll tell you yes or no."
                        },
                        {
                          "name": "!selfie",
                          "value": "I'll send you a random picture of me."
                        },
                        {
                          "name": "!cat",
                          "value": "Random cat pictures from Tumblr."
                        },
                        {
                          "name": "!thispersondoesnotexist",
                          "value": "Generates a person that does not exist."
                        },
                        {
                          "name": "!waifu",
                          "value": "Generates a waifu for you to love. <3"
                        },
                        {
                          "name": "!random",
                          "value": "One command sent at random..."
                        },
                        {
                          "name": "!lastNum",
                          "value": "I'll tell you the number of the last gif I sent. (Number 94 of an anime gif would equal anime_94)"
                        },
                        {
                          "name": "!birthdayReset",
                          "value": "Reset the cooldown on birthday messages."
                        },
                        {
                          "name": "!who is my \"Family Member Here\"",
                          "value": "I will reveal some of my relations..."
                        },
                        {
                          "name": "!help",
                          "value": "You just did that..."
                        }
                      ]
                    }
              };
              channel.send(data);
          break;
      }
  }
  return lastNum
}

module.exports.command = command;

File: commands.py
import discord
from re import search
import random
from random import randrange
from random import randint
from loggingChannel import sendLog
import os, os.path
import requests
import json
from dotenv import load_dotenv
from fileManager import sendImage


async def sendGif(client, channel, search_term, random):
    # gif start
    load_dotenv()
    apikey = os.getenv('TENOR_API_KEY')
    lmt = 50
    url = "https://api.tenor.com/v1/search?q="
    if random:
        url = "https://api.tenor.com/v1/random?q="
        lmt = 1
    r = requests.get(url + ("%s&key=%s&limit=%s" % (search_term, apikey, lmt)))
    if r.status_code == 200:
        # load the GIFs using the urls for the smaller GIF sizes
        top_gifs = json.loads(r.content)
        # print(top_gifs)
        selected_gif = top_gifs['results'][randrange(lmt)]
        print(await sendLog(log=("Gif selected " + selected_gif["url"]), client=client))
        await channel.send(selected_gif["url"])
    else:
        top_8gifs = None


async def checkForCommands(message, client):
    if search("^(anime)", message.content.lower()):
        localOrOnline = randint(1, 3)
        if localOrOnline >= 2:  # Online
            print(await sendLog(
                log=f'{message.author.name} has requested anime! Sending online gif!',
                client=client))
            await sendGif(client, message.channel, "cute anime girl", random=False)
        else:
            DIR = './images/anime/'
            options = len([name for name in os.listdir(DIR) if os.path.isfile(os.path.join(DIR, name))])
            animeNum = random.randint(0, (options - 1))
            print(await sendLog(
                log=f'{message.author.name} has requested anime! Sending anime#{animeNum}!',
                client=client))
            await sendImage(message, client, "anime_", animeNum, DIR)

    if search("^(selfie)", message.content.lower()):
        DIR = './images/selfies/'
        options = len([name for name in os.listdir(DIR) if os.path.isfile(os.path.join(DIR, name))])
        selfieNum = random.randint(0, (options - 1))
        print(await sendLog(
            log=f'{message.author.name} has asked for a selfie. Sending -> #{selfieNum}!',
            client=client))
        await sendImage(message, client, "selfie_", selfieNum, DIR)

    if search("roll(_|\s)([0-9]+)d([0-9]+)", message.content.lower()):
        value = 0
        dice = [int(i) for i in message.content.split() if i.isdigit()]

        print(len(dice))
        if len(dice) == 2:
            for i in range(dice[0]):
                value = value + dice[1]
            embed = discord.Embed(colour=discord.Colour(0x259944), url="https://discordapp.com",
                                  description=f'You rolled {value} on your {dice[0]}d{dice[1]}. Good job! At the very least you get an A+ for effort so isn\'t that nice.')
            embed.set_thumbnail(url="https://gilkalai.files.wordpress.com/2017/09/dice.png?w=640")
            embed.set_author(name="Steve from Accounting", url="https://www.google.com/error",
                             icon_url="https://www.topaccountingdegrees.org/wp-content/uploads/2015/08/Accounting-7.jpg")
            await message.channel.send(embed=embed)

    # Sins
    if search("^(sins)", message.content.lower()):
        await message.channel.send('[THIS WILL SHOW YOU PEOPLES SINS BUT IT\'S UNDER CONSTRUCTION]')

    # Punch
    if search("^(punch)", message.content.lower()):
        await message.channel.send('[THIS WILL PUNCH PEOPLE BUT IT\'S UNDER CONSTRUCTION]')

    # Grank
    # if search("^(grank)", message.content.lower()):
    # await message.channel.send('[THIS WILL SHOW RANK BUT IT\'S UNDER CONSTRUCTION]')

    if search("^(mcinfo)", message.content.lower()):
        url = "https://api.ipify.org/?format=json"
        ip = "error"
        r = requests.get(url)
        if r.status_code == 200:
            ip = json.loads(r.content)
            print(f'IP: {ip["ip"]}')
        await message.channel.send(f'Minecraft Server IP: {ip["ip"]}:26969')

    if search("^(ping)", message.content.lower()):
        await message.channel.send('What are you expecting? Me to say pong back?')

    if search("^(wtf)", message.content.lower()):
        await message.channel.send('Rude!')

    if search("^(nani)", message.content.lower()):
        await message.channel.send('何')

    if search("^(thispersondoesnotexist|tpdne)", message.content.lower()):
        url = "https://fakeface.rest/face/json"
        r = requests.get(url)
        if r.status_code == 200:
            thispersondoesnotexist = json.loads(r.content)
            # print(top_gifs)
            await message.channel.send(thispersondoesnotexist['image_url'])

    if search("^(waifu)", message.content.lower()):
        url = "https://www.thiswaifudoesnotexist.net/example-"
        num = random.randint(0, 100000)
        await message.channel.send(url + str(num) + ".jpg")

    if search("^((id)|(fake(\s|_|)id))", message.content.lower()):
        url = "https://random-user.p.rapidapi.com/getuser"

        headers = {
            'x-rapidapi-key': os.getenv('x-rapidapi-key'),
            'x-rapidapi-host': "random-user.p.rapidapi.com"
        }

        r = requests.get(url, headers=headers)
        if r.status_code == 200:
            id = json.loads(r.content)
            id = id['results'][0]
            print(id)
            embed = discord.Embed(colour=discord.Colour(0xb8b8b8), description=f'ID for {id["name"]["first"]} {id["name"]["last"]}:')

            embed.set_thumbnail(url=id["picture"]["large"])
            embed.set_author(name=f'{id["name"]["title"]} {id["name"]["first"]} {id["name"]["last"]}',
                             icon_url=id["picture"]["large"], url=id["picture"]["large"])
            location = id["location"]
            embed.add_field(name="Location", value=f'{location["street"]["number"]} {location["street"]["name"]} {location["city"]}, {location["state"]}, {location["country"]}, {location["postcode"]}')
            embed.add_field(name="Email", value=f'Email: {id["email"]} \nUsername: {id["login"]["username"]} \nPassword: {id["login"]["password"]}')
            embed.add_field(name="Phone Number", value=f'{id["cell"]}')
            embed.add_field(name="Date of Birth", value=f'Age: {id["dob"]["age"]} DOB: {id["dob"]["date"]}')
            if id['id']['value'] == id['id']['value'] and id['id']['value'] is not None:
                embed.add_field(name="ID", value=f'Type: {id["id"]["name"]} Value: {id["id"]["value"]}')

            await message.channel.send(embed=embed)

    if search("^(hot)", message.content.lower()):
        file = discord.File("./video/Hot.mp4", filename="hot.mp4")
        await message.channel.send(file=file)

    if search("^(joke)", message.content.lower()):
        num = random.randint(0, 3)
        if num >= 1:
            print("Sending joke from joke3.p.rapidapi")
            url = "https://joke3.p.rapidapi.com/v1/joke"
            headers = {
                'x-rapidapi-key': "dc02b90e2emsh6be8145fc6dd65ep198267jsnf83b7e473ef7",
                'x-rapidapi-host': "joke3.p.rapidapi.com"
            }
            r = requests.get(url, headers=headers)
            if r.status_code == 200:
                joke = json.loads(r.content)
                await message.channel.send(joke['content'])
        else:
            print("Sending joke from sv443.net/jokeapi")
            url = "https://sv443.net/jokeapi/v2/joke/Any?blacklistFlags=racist&type=single"
            r = requests.get(url)
            if r.status_code == 200:
                joke = json.loads(r.content)
                await message.channel.send(joke['joke'])

    if search("^(insult)", message.content.lower()):
        url = "https://evilinsult.com/generate_insult.php?lang=en&type=json"
        r = requests.get(url)
        if r.status_code == 200:
            insult = json.loads(r.content)
            await message.channel.send(insult['insult'])

    if search("^(fact)", message.content.lower()):
        url = "https://uselessfacts.jsph.pl/random.json?language=en"
        r = requests.get(url)
        if r.status_code == 200:
            fact = json.loads(r.content)
            await message.channel.send(fact['text'])

    if search("^(advi(s|c)e)", message.content.lower()):
        url = "https://api.adviceslip.com/advice"
        r = requests.get(url)
        if r.status_code == 200:
            advice = json.loads(r.content)
            await message.channel.send(advice["slip"]['advice'])

    if search("^(cat|kitty)", message.content.lower()):
        cat_api = os.getenv('THE_CAT_API')
        url = "https://api.thecatapi.com/v1/images/search?api_key=" + cat_api
        r = requests.get(url)
        if r.status_code == 200:
            cat = json.loads(r.content)
            await message.channel.send(cat[0]["url"])

    if search("^(dog|pup)", message.content.lower()):
        url = "https://random.dog/woof.json?ref=apilist.fun"
        r = requests.get(url)
        if r.status_code == 200:
            dog = json.loads(r.content)
            await message.channel.send(dog["url"])

    if search("^(yesorno)", message.content.lower()):
        url = "https://yesno.wtf/api"
        r = requests.get(url)
        if r.status_code == 200:
            yesorno = json.loads(r.content)
            await message.channel.send(yesorno['image'])
            await message.channel.send(yesorno['answer'].capitalize())

    if search("^(zerotier)", message.content.lower()):
        await message.channel.send('[THIS WILL SHOW YOU HOW TO SETUP ZERO TIER BUT IT\'S UNDER CONSTRUCTION]')

    if search("^(r2loadout)", message.content.lower()):
        from r2loadout import get_r2loadout
        await message.channel.send(embed=get_r2loadout())

    if search("^(random)", message.content.lower()):
        await message.channel.send('[THIS WILL SEND A RANDOM COMMAND BUT IT\'S UNDER CONSTRUCTION]')

    if search("^(help)", message.content.lower()):
        embed = discord.Embed(title="Help", colour=discord.Colour(0x9b9b9b), url="https://discordapp.com",
                              description="This is a list of commands you can use with Geoffery's Son."
                                          "\n\n**OwO**\nConfuses me."
                                          "\n\n**Is it possible to learn this power?**\nNo, leave us..."
                                          "\n\n**The sun is a deadly laser!**\nNot anymore there's a blanket!"
                                          "\n\n**10th/9th time!**\n10th/9th time!"
                                          "\n\n**Sauce**\nThe work of the devil."
                                          "\n\n**Heresy**\nWe must deal with is immediately!"
                                          "\n\n**Ravioli ravioli**\nDragon Loli"
                                          "\n\n**Hentai**\nWait that's illegal!"
                                          "\n\n**Hello there**\nGeneral Kenobi!"
                                          "\n\n**Trap**\nWhat do you think?"
                                          "\n\n**!anime**\nAnime gifs for everyone!"
                                          "\n\n**!roll 1d6**\nRolls dice based on what you send. For example !roll 3d2 will roll 3 d2 dice and than send the result."
                                          "\n\n**!sins @yourfiends**\nThis where inform you of the sins of your friends. I'd watch out for Steve from accounting..."
                                          "\n\n**!punch @yourfiends**\nPunch your friends over the internet from a safe distance."
                                          "\n\n**!mcinfo**\nInfo on the Minecraft Server if one is running."
                                          "\n\n**!ping**\nWhat do you expect?"
                                          "\n\n**!wtf**\nWhy is this a command?"
                                          "\n\n**!nani**\nIt can translate to weeb characters."
                                          "\n\n**!thispersondoesnotexist or !tpdne**\nThe smart ai over at [thispersondoesnotexist](https://thispersondoesnotexist.com) will send us a face that does not exist."
                                          "\n\n**!waifu**\nThe smart ai over at [thiswaifudoesnotexist](https://www.thiswaifudoesnotexist.net) will send us a waifu that does not exist."
                                          "\n\n**!id or !fake id**\nThis will create your new identity."
                                          "\n\n**!hot**\nBRRRRRRRR!!"
                                          "\n\n**!joke**\nI'll tell you a joke!"
                                          "\n\n**!insult**\nI'll insult you! Be prepared some of these are pretty terrible..."
                                          "\n\n**!fact**\nI'll tell you a random fact."
                                          "\n\n**!advice**\nI'll give you some advice."
                                          "\n\n**!cat or !dog**\nI'll send a random image of a kitty or dogo!"
                                          "\n\n**!yesorno**\nI'll tell you if the answer is yes or no."
                                          "\n\n**!r2loadout**\nA random load out for a Risk of Rain 2 Command run. I hope Rnjesus is on your side."
                                          "\n\n**!help**\nConsidering your already here you probably already know this one...")

        embed.set_author(name="Geoffery's Son", url="https://github.com/Geoffery10/Geoffery-s-Son-Discord-Bot",
                         icon_url="https://github.com/Geoffery10/Geoffery-s-Son-Discord-Bot/blob/Python/images/selfies/selfie_04.png?raw=true")
        await message.channel.send(embed=embed)